astropy/cosmology/tests/test_cosmology.py

```diff
@@ -1577,6 +1577,11 @@ def test_z_at_value():
     assert allclose(z_at_value(cosmo.distmod, 46 * u.mag),
                     1.9913891680278133, rtol=1e-6)
 
+    # Test array:
+    assert allclose(z_at_value(cosmo.distmod, np.array([46]*4) * u.mag,
+                               zmin=1.5, zmax=2.5, nbins=10000),
+                    [1.9913891680278133]*4, rtol=1e-3)
```

Comment:
I think Numpy is smart enough to auto broadcast.
```suggestion
                    1.9913891680278133, rtol=1e-3)
```